sdk/python/kfp/dsl/component_factory.py

```diff
@@ -147,7 +147,7 @@ def _get_packages_to_install_command(
                 install_parts=[
                     f'kfp=={kfp.__version__}',
                     '--no-deps',
-                    'typing-extensions>=3.7.4,<5; python_version<"3.9"',
+                    'typing-extensions>=3.7.4,<5; python_version<"3.13"',
```

Comment:
Can you explain why this is required? It seems typing extensions shouldn't be needed in versions greater than 3.9, but this says it is needed in Python 3.9-3.12 inclusive.